src/audio/codec_adapter/codec/cadence.c

```diff
@@ -172,6 +172,47 @@ int cadence_codec_init(struct comp_dev *dev)
 	return ret;
 }
 
+static int update_stream_params(struct comp_dev *dev)
+{
+	int ret = 0;
+	struct comp_data *ca_data = comp_get_drvdata(dev);
+	struct codec_data *codec = comp_get_codec(dev);
+	struct cadence_codec_data *cd = codec->private;
+	struct sof_ipc_stream_params *stream = &ca_data->stream_params;
+	/* copy stream params localy to ensure container size is correct */
+	uint32_t sample_width = (stream->frame_fmt == SOF_IPC_FRAME_S16_LE) ? 16 : 32;
+	uint32_t sample_rate = stream->rate;
+	uint32_t channels = stream->channels;
+
+	comp_dbg(dev, "update_stream_params() start");
+
+	/* update stream parameters */
+	/* sample rate */
+	API_CALL(cd, XA_API_CMD_SET_CONFIG_PARAM, cd->sample_rate_id,
+		 &sample_rate, ret);
+	if (LIB_IS_FATAL_ERROR(ret)) {
+		comp_err(dev, "update_stream_params(): failed to set sample rate.");
+		goto err;
+	}
+	/* sample width */
+	API_CALL(cd, XA_API_CMD_SET_CONFIG_PARAM, cd->sample_width_id,
+		 &sample_width, ret);
+	if (LIB_IS_FATAL_ERROR(ret)) {
+		comp_err(dev, "update_stream_params(): failed to set sample width.");
+		goto err;
+	}
+	/* number of channels */
+	API_CALL(cd, XA_API_CMD_SET_CONFIG_PARAM, cd->channels_id,
+		 &channels, ret);
+	if (LIB_IS_FATAL_ERROR(ret)) {
+		comp_err(dev, "update_stream_params(): failed to set channel count.");
+		goto err;
+	}
+err:
+	comp_dbg(dev, "update_stream_params() done, returned %d", ret);
+	return ret;
+}
```

Comment:
If we get here I think we should `return 0`. For example setting channels or sample rate for MP3 is a non fatal error. MP3 decoder only needs the `sample width` the rest it will find by itself parsing the mp3 header.

Reply:
@dbaluta, I have just updated, I think this new way of handling update_stream_params() will cover fatal & non-fatal cases.

Reply:
@dbaluta is this resolved, sorry I cant tell from the diff.